tests/core_tests.py

```diff
@@ -726,7 +727,23 @@ def test_user_profile(self, username='admin'):
         data = self.get_json_resp('/superset/fave_dashboards_by_username/{}/'.format(username))
         self.assertNotIn('message', data)
 
+    def test_slice_id_is_always_logged_correctly_on_web_request(self,):
+        # superset/explore case
+        slc = db.session.query(models.Slice).filter_by(slice_name='Girls').one()
+        db.session.expunge_all()
+        self.get_resp(slc.slice_url)
+        qry = db.session.query(models.Log).filter_by(slice_id=slc.id, )
+        self.assertEqual(1, qry.count())
+
+    def test_slice_id_is_always_logged_correctly_on_ajax_request(self,):
+        # superset/explore_json case
+        self.login(username="admin")
+        slc = db.session.query(models.Slice).filter_by(slice_name='Girls').one()
+        qry = db.session.query(models.Log).filter_by(slice_id=slc.id, )
```

Comment:
NIT: I'd move this to right above the `self.assertEqual(1, qry.count())` line as the variable doesn't need to exist while the request takes place. Shorter lived variables are better in general. It's pretty minor but could cause confusion as to when the query is actually executed.